tensorboard/webapp/widgets/line_chart_v2/lib/worker_pool_test.ts

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {WorkerPool} from './worker_pool';
+
+describe('line_chart_v2/lib/worker_pool', () => {
+  let workerFactory: jasmine.Spy;
+
+  beforeEach(() => {
+    workerFactory = jasmine.createSpy().and.callFake(() => {
+      return jasmine.createSpyObj(Worker, ['postMessage']);
+    });
+  });
+
+  it('returns a worker like that it can postMessage to', () => {
+    const workerSpy = jasmine.createSpyObj(Worker, ['postMessage']);
+    workerFactory.and.returnValue(workerSpy);
+
+    const workerLike = new WorkerPool('testing', 2, workerFactory).getNext();
+    workerLike.postMessage('foo', []);
+
+    expect(workerFactory).toHaveBeenCalledTimes(1);
+    expect(workerSpy.postMessage).toHaveBeenCalledTimes(1);
+    expect(workerSpy.postMessage).toHaveBeenCalledWith('foo', []);
+  });
+
+  describe('allocation', () => {
+    it('allocates pools even when instantiated pool with the same path', () => {
+      new WorkerPool('a', 1, workerFactory).getNext();
+      new WorkerPool('a', 1, workerFactory).getNext();
+      expect(workerFactory).toHaveBeenCalledTimes(2);
+    });
+
+    it('allocates upto maxPoolSize then allocates from first one again', () => {
+      const pool = new WorkerPool('a', 3, workerFactory);
+      const workers = [...new Array(5)].map(() => {
```

Comment:
…wow, TIL that `new Array(5)` is not the same as `[...new Array(5)]`.
Like, I knew about empty slots, but I always did `Array(5).fill().map`.

How lovely that `[...xs]` and `xs.slice()` are in fact _not_ always
equivalent when `Array.isArray(xs)`.